ddcloud/resource_server.go

```diff
@@ -888,12 +891,36 @@ func findPublicIPv4Address(apiClient *compute.Client, networkDomainID string, pr
 }
 
 func validateAdminPassword(adminPassword string, image compute.Image) error {
+	validPassword := true
+	lettersCount, numCount, specialCount, upperCaseCount := 0, 0, 0, 0
+
+	for _, s := range adminPassword {
+		switch {
+		case unicode.IsDigit(s):
+			numCount++
+		case unicode.IsUpper(s):
+			lettersCount++
+			upperCaseCount++
+		case unicode.IsPunct(s) || unicode.IsSymbol(s):
+			specialCount++
+		case unicode.IsLetter(s) || s == ' ':
+			lettersCount++
+		default:
+			//
+		}
+	}
+	if (numCount < 1) || (lettersCount <= 8) || (specialCount < 1) || (upperCaseCount < 1) {
+		validPassword = false
+	}
+
 	switch image.GetType() {
+
 	case compute.ImageTypeOS:
 		// Admin password is always mandatory for OS images.
-		if adminPassword == "" {
-			return fmt.Errorf("Must specify an initial admin password when deploying an OS image")
+		if !validPassword {
+			return fmt.Errorf("password does not meet complexity requirements. Needs 9 characters, 1 upper, 1 lower, 1 number and a special char")
```

Comment:
Can you still mention that this is only required when deploying OS images and some types of Windows customer images (as appropriate in each section?)... Just be sure that the password is only mandatory for those cases...